client/asset/dcr/dcr.go

```diff
@@ -754,17 +779,154 @@ func (dcr *ExchangeWallet) PreSwap(req *asset.PreSwapForm) (*asset.PreSwap, erro
+	for maxBump = 2.0; maxBump > 1.01; maxBump -= 0.1 {
+		tryEst, splitUsed, _, err := dcr.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, split, maxBump)
+		// If the split used wasn't the configured value, this option is not
+		// available.
+		if err == nil && split == splitUsed {
+			maxBumpEst = tryEst
+			break
+		}
+	}
+
+	if maxBumpEst != nil {
+		noBumpEst, _, _, err := dcr.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, split, 1.0)
+		if err != nil {
+			// shouldn't be possible, since we already succeded with a higher bump.
+			return nil, fmt.Errorf("error getting no-bump estimate: %w", err)
+		}
+
+		bumpLabel := "2X"
+		if maxBump < 2.0 {
+			bumpLabel = strconv.FormatFloat(maxBump, 'f', 1, 64) + "X"
+		}
+
+		extraFees := maxBumpEst.RealisticWorstCase - noBumpEst.RealisticWorstCase
+		desc := fmt.Sprintf("Bump fees up to %.1fx (up to ~%s DCR more) for faster settlement when network traffic is high.",
+			maxBump, amount(extraFees))
+
+		opts = append(opts, &asset.OrderOption{
+			ConfigOption: asset.ConfigOption{
+				Key:          swapFeeBumpKey,
+				DisplayName:  "Faster Swaps",
+				Description:  desc,
+				DefaultValue: 1.0,
+			},
+			XYRange: &asset.XYRange{
+				Start: asset.XYRangePoint{
+					Label: "1X",
+					X:     1.0,
+					Y:     float64(req.FeeSuggestion),
+				},
+				End: asset.XYRangePoint{
+					Label: bumpLabel,
+					X:     maxBump,
+					Y:     float64(req.FeeSuggestion) * maxBump,
+				},
+				XUnit: "X",
+				YUnit: "atoms/B",
+			},
+		})
+	}
+
 	return &asset.PreSwap{
 		Estimate: est,
+		Options:  opts,
 	}, nil
 }
 
+// splitOption constructs an *asset.OrderOption with customized text based on the
+// difference in fees between the configured and test split condition.
+func (dcr *ExchangeWallet) splitOption(req *asset.PreSwapForm, utxos []*compositeUTXO, bump float64) *asset.OrderOption {
+	noSplitEst, _, noSplitLocked, err := dcr.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, false, bump)
+	if err != nil {
+		dcr.log.Errorf("estimateSwap (no split) error: %v", err)
+		return nil
+	}
+	splitEst, splitUsed, splitLocked, err := dcr.estimateSwap(req.Lots, req.LotSize, req.FeeSuggestion, utxos, req.AssetConfig, true, bump)
+	if err != nil {
+		dcr.log.Errorf("estimateSwap (with split) error: %v", err)
+		return nil
+	}
+	if !splitUsed {
+		// unable to do the split. no option.
+		dcr.log.Debugf("split option unavailable")
+		return nil
+	}
+
+	xtraFees := splitEst.RealisticWorstCase - noSplitEst.RealisticWorstCase
+	pctChange := (float64(splitEst.RealisticWorstCase)/float64(noSplitEst.RealisticWorstCase) - 1) * 100
+	overlock := noSplitLocked - splitLocked
+
+	var reason string
+	if pctChange > 1 {
+		reason = fmt.Sprintf("+%d%% fees, -%s DCR overlock", int(math.Round(pctChange)), amount(overlock))
+	} else {
+		reason = fmt.Sprintf("+%.1f%% fees, -%s DCR overlock", pctChange, amount(overlock))
```

Comment:
I find the negative amount overlock a bit confusing.  We are trying to say that they are paying more in fees because of the added transaction, but doing that eliminates/avoids the over locking of %s DCR.
Is there are clearer way to express this within the constraints of this ui dialog?